src/publisher/middleware.py

```diff
@@ -76,61 +32,19 @@ def flatten_accept(header, just_elife=False):
     return lst
 
 
+def has_structured_abstract(ajson):
+    "returns True if ajson contains a structured abstract"
+    if "abstract" in ajson:
+        # a regular abstract is just the keys 'type' and 'text'
+        # a structured abstract also has an 'id' field
+        return "id" in ajson["abstract"]["content"][0]
```

Comment:
It is possible (which I think was the plan) that the `id` is an optional attribute for an abstract section I requested, so if the API schema is still that way, then a structured abstract may not have any `id` values.

Reply:
nice catch. I'll take a second look at the structured abstract spec

Reply:
[here it is in our spec](https://github.com/elifesciences/api-raml/commit/93bda43a661c71c19468a936880793363a4205bd#diff-5095a45df81a60f9049fd5ce3545a678R8-R26)

It's not 100% clear to me, but it reads as if: an abstract is either ('`oneOf`') a paragraph type or a map with the `type`, `id`, `title` and `content` keys ('`properties`'). There is no `required` section listing the required properties and none of the properties have `required: True` against them and  [the default value for `required` appears to be `True`](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#property-declarations).

So we should be safe with just checking for an `id` value. @gnott , I hate to be a pain but could you chase down where you heard that `id` should be optional? It might have been a requirement that was overlooked, or a requirement that was dropped or I've completely misread the spec (cc @nlisgo )

Reply:
Here I think is my first comment/request for adding an optional `@id` attribute, however in the end after the changes to the schema, I didn't check whether it is still optional https://github.com/elifesciences/api-raml/pull/246#discussion_r434232366

Whether the `@id` attribute will always be there for eLife articles would be something to run by Production and the typesetter I think, but conceivably a structured abstract could omit any `@id` from the `<sec>` tags and be acceptable.

Reply:
it looks like a requirement that was overlooked then (if I'm not misreading the spec). @nlisgo to please confirm.

Reply:
after our chat, I'll switch this to look for `type` rather than `id`. I expect to merge this Thursday and then the api-raml changes made after this PR makes it into production

Reply:
update: turns out 'type' is shared with regular abstracts as well. I've changed the test so that a structured abstract is simply missing the 'text' key.